README.md

```diff
@@ -276,8 +280,19 @@ matching Kubernetes versions are listed below:
 | release-0.15      | Kubernetes 1.15 branch v1.15.x | unsupported |
 | release-0.11      | Kubernetes 1.11 branch v1.11.x | unsupported |
 
-[Go environment]: https://golang.org/doc/install
-[Kubernetes cluster]: https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/
+*Note:* Device plugins leverage the Kubernetes v1 API. The API itself is GA (generally available) and doesn't change between Kubernetes versions. One doesn't necessarily need use the latest Kubernetes cluster with the latest device plugin version. Using a newer device plugins release should work without issues on an older Kubernetes cluster. One possible exception to this are the device plugins CRDs that can vary between versions.
+
+[Go environment](https://golang.org/doc/install) | [Kubernetes cluster setup](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/)
+
+### Release procedures
+
+Project's release cadence is tied to Kubernetes release cadence. Device plugins release typically follows a couple of weeks after the Kubernetes release. There can be some delays on the releases due to required changes in the pull request pipeline. Once the content is available in the `main` branch and CI & e2e validation <span style="color:green">PASS</span>es, release branch will be created (e.g. release-0.26). The HEAD of release branch will also be tagged with the corresponding [tag](https://github.com/intel/intel-device-plugins-for-kubernetes/tags) (e.g. v0.26.0).
+
+During the [release creation](https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1393), the project's documentation, deployment files etc. will be [changed](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1441) to point to the newly created version.
+
+Pull requests that are required for specific release are labeled with the corresponding [milestone](https://github.com/intel/intel-device-plugins-for-kubernetes/milestones).
+
+Patch releases (e.g. 0.26.3) are done on an adhoc basis if there are security issues or minor fixes requested for specific version.
```

Comment:
Typo fixes:
```suggestion
Patch releases (e.g. 0.26.3) are done on an ad-hoc basis if there are security issues or minor fixes are requested for a specific version.
```

Reply:
we should probably say 'on a need basis' rather than 'ad-hoc'